CI: operator/pkg/bgp: flake in TestRouterIDAllocation: the object has been modified; resource version mismatch

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/operator/pkg/bgp   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestRouterIDAllocation (3.12s)
--- FAIL: TestRouterIDAllocation/multiple_nodes_and_multiple_instances_with_single_cluster_configs_and_cleanup_the_2_instances_in_the_cluster_config (1.20s)

    logger.go:256: time=2026-02-19T08:39:14.258Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:385 msg="Starting hive"
    logger.go:256: time=2026-02-19T08:39:14.259Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:389 msg="Started hive" duration=45.779µs
    logger.go:256: time=2026-02-19T08:39:14.459Z level=INFO source=/home/runner/work/cilium/cilium/operator/pkg/bgp/manager.go:151 msg="BGP control plane operator started" module=bgp-cp-operator
    cluster_test.go:1637: 
        	Error Trace:	/home/runner/work/cilium/cilium/operator/pkg/bgp/cluster_test.go:1637
        	            				/home/runner/work/cilium/cilium/operator/pkg/bgp/cluster_test.go:1577
        	            				/opt/hostedtoolcache/go/1.26.0/x64/src/runtime/asm_amd64.s:1771
        	Error:      	Received unexpected error:
        	            	Operation cannot be fulfilled on ciliumbgpclusterconfigs.cilium.io "test-cc-multi-node": the object has been modified; resource version mismatch, current "18", got "15"
        	Test:       	TestRouterIDAllocation/multiple_nodes_and_multiple_instances_with_single_cluster_configs_and_cleanup_the_2_instances_in_the_cluster_config
    logger.go:256: time=2026-02-19T08:39:15.461Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:401 msg="Stopping hive"
    logger.go:256: time=2026-02-19T08:39:15.461Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:405 msg="Stopped hive" duration=601.289µs
```

https://github.com/cilium/cilium/actions/runs/22174214293/job/64118716945

Can easily be reproduced locally using `go test -c ./operator/pkg/bgp && stress ./bgp.test`.

cc @cilium/sig-bgp

--- a/operator/pkg/bgp/cluster_test.go
+++ b/operator/pkg/bgp/cluster_test.go
@@ -316,7 +316,7 @@ func Test_NodeLabels(t *testing.T) {
 			upsertNode(req, ctx, f, tt.node)
 
 			// upsert BGP cluster config
-			upsertBGPCC(req, ctx, f, tt.clusterConfig)
+			req.NoError(upsertBGPCC(ctx, f, tt.clusterConfig))
 
 			// validate node configs
 			assert.EventuallyWithT(t, func(c *assert.CollectT) {
@@ -636,7 +636,7 @@ func Test_ClusterConfigSteps(t *testing.T) {
 			}
 
 			// upsert BGP cluster config
-			upsertBGPCC(req, ctx, f, step.clusterConfig)
+			req.NoError(upsertBGPCC(ctx, f, step.clusterConfig))
 
 			// upsert node overrides
 			for _, nodeOverride := range step.nodeOverrides {
@@ -851,7 +851,8 @@ func TestClusterConfigConditions(t *testing.T) {
 
 			// Setup resources
 			upsertNode(req, ctx, f, &node)
-			upsertBGPCC(req, ctx, f, tt.clusterConfig)
+			req.NoError(upsertBGPCC(ctx, f, tt.clusterConfig))
+
 			upsertBGPPC(req, ctx, f, &peerConfig)
 
 			require.EventuallyWithT(t, func(ct *assert.CollectT) {
@@ -1108,7 +1109,7 @@ func TestConflictingClusterConfigCondition(t *testing.T) {
 						},
 					},
 				}
-				upsertBGPCC(req, ctx, f, clusterConfig)
+				req.NoError(upsertBGPCC(ctx, f, clusterConfig))
 			}
 
 			require.EventuallyWithT(t, func(ct *assert.CollectT) {
@@ -1219,7 +1220,7 @@ func TestDisableClusterConfigStatusReport(t *testing.T) {
 	}
 
 	// Setup resourses with status
-	upsertBGPCC(req, ctx, f, clusterConfig)
+	req.NoError(upsertBGPCC(ctx, f, clusterConfig))
 
 	// Wait for status to be cleared
 	require.EventuallyWithT(t, func(ct *assert.CollectT) {
@@ -1532,7 +1533,7 @@ func TestRouterIDAllocation(t *testing.T) {
 			}
 			for _, clusterConfig := range tt.InitClusterConfigs {
 				config := clusterConfig
-				upsertBGPCC(req, ctx, f, config)
+				req.NoError(upsertBGPCC(ctx, f, config))
 			}
 
 			if tt.nodeOverrides != nil {
@@ -1574,7 +1575,9 @@ func TestRouterIDAllocation(t *testing.T) {
 				// version even after receiving an event for the new version.
 				if tt.FinalClusterConfigs != nil {
 					for _, clusterConfig := range tt.FinalClusterConfigs {
-						upsertBGPCC(req, ctx, f, clusterConfig)
+						if err := upsertBGPCC(ctx, f, clusterConfig); !assert.NoError(c, err) {
+							return
+						}
 					}
 				}
 				NodeConfigs, err := f.bgpnClient.List(ctx, meta_v1.ListOptions{})
@@ -1619,22 +1622,22 @@ func upsertNode(req *require.Assertions, ctx context.Context, f *fixture, node *
 	req.NoError(err)
 }
 
-func upsertBGPCC(req *require.Assertions, ctx context.Context, f *fixture, bgpcc *v2.CiliumBGPClusterConfig) {
+func upsertBGPCC(ctx context.Context, f *fixture, bgpcc *v2.CiliumBGPClusterConfig) error {
 	if bgpcc == nil {
-		return
+		return nil
 	}
 
 	existing, err := f.bgpcClient.Get(ctx, bgpcc.Name, meta_v1.GetOptions{})
 	if err != nil && k8sErrors.IsNotFound(err) {
 		_, err = f.bgpcClient.Create(ctx, bgpcc, meta_v1.CreateOptions{})
 	} else if err != nil {
-		req.Fail(err.Error())
+		return err
 	} else {
 		bgpcc.SetUID(existing.GetUID())
 		bgpcc.SetResourceVersion(existing.GetResourceVersion())
 		_, err = f.bgpcClient.Update(ctx, bgpcc, meta_v1.UpdateOptions{})
 	}
-	req.NoError(err)
+	return err
 }
 
 func upsertBGPPC(req *require.Assertions, ctx context.Context, f *fixture, bgppc *v2.CiliumBGPPeerConfig) {

